### PR TITLE
[AppBar] Add API to auto adjust height based on HeaderStackView for app bar.

### PR DESCRIPTION
**Pull request by fluxminds**
Part of https://github.com/material-components/material-components-ios/issues/8677.

This API helps to reduce the complexity of calculating minimumHeight and maximumHeight on FlexibleHeader when user wants to update the app bar height based on header stack view.

This flag is required to be set during configuration stage. The reset behavior (turning back to NO) was not implemented because it would require adding API to minMaxHeight class and all configuration needs be to be set before flexible header is used. This feature can be added if needed.

### CODE REVIEW

**Comment by fluxminds** [comment]
Discussed with @jverkoey offline. Revised the API name and ready for review.

File: components/AppBar/src/MDCAppBarViewController.h [hunk 1]
```diff
@@ -36,6 +36,14 @@
  */
 @property(nonatomic, strong, nonnull) MDCHeaderStackView *headerStackView;
 
+/**
+ When this flag is set to YES, the height of the app bar will be automatically adjusted to the sum
+ of the top bar height and the bottom bar height.
```
Comment (jverkoey) [inline components/AppBar/src/MDCAppBarViewController.h:41]: Suggested docs:

```
Enabling this property will disable `minMaxHeightIncludesSafeArea` on the flexible header view.
```

File: components/AppBar/src/MDCAppBarViewController.m [hunk 2]
```diff
@@ -276,6 +288,15 @@ - (void)dismissSelf {
   }
 }
 
+#pragma mark - Private
+
+- (void)adjustHeightBasedOnHeaderStackView {
+  CGFloat heightSum = 0;
+  heightSum += [self.headerStackView.topBar sizeThatFits:self.view.frame.size].height;
+  heightSum += [self.headerStackView.bottomBar sizeThatFits:self.view.frame.size].height;
```
Comment (jverkoey) [inline components/AppBar/src/MDCAppBarViewController.m]: Always use `self.view.bounds` when calculating dimensions of subviews.

```objc
  heightSum += [self.headerStackView.topBar sizeThatFits:self.view.bounds.size].height;
  heightSum += [self.headerStackView.bottomBar sizeThatFits:self.view.bounds.size].height;
```

File: components/AppBar/src/MDCAppBarViewController.m [hunk 3]
```diff
@@ -276,6 +288,15 @@ - (void)dismissSelf {
...
+  self.headerView.minimumHeight = self.headerView.maximumHeight = heightSum;
```
Comment (jverkoey) [inline components/AppBar/src/MDCAppBarViewController.m]: Readability:

```objc
self.headerView.minimumHeight = heightSum;
self.headerView.maximumHeight = heightSum;
```

File: components/AppBar/tests/unit/MDCAppBarViewControllerTests.m [hunk 4]
```diff
@@ -89,4 +89,22 @@ - (void)testHeaderViewElevationSetByShadowIntensityChange {
...
+  appBarController.headerStackView.bottomBar = bottomBar;
+
+  // When
+  appBarController.shouldAdjustHeightBasedOnHeaderStackView = YES;
+
+  // Then
+  CGFloat minHeight = appBarController.headerView.minimumHeight;
+  CGFloat maxHeight = appBarController.headerView.maximumHeight;
+  XCTAssertEqual(minHeight, maxHeight);
+  XCTAssertEqualWithAccuracy(minHeight, 200 + navigationBarHeight, 0.01);
```
Comment (jverkoey) [inline components/AppBar/tests/unit/MDCAppBarViewControllerTests.m]: Pull 200 out to a constant in the `Given` section to clarify that it's semantically the same value here and above.

**Review by jverkoey — APPROVED** [review]

File: components/AppBar/src/MDCAppBarViewController.m (hunk 2 above)
Comment (fluxminds) [reply components/AppBar/src/MDCAppBarViewController.m]: Addressed in 1b97d553ff0a58d5597bbe7382a338085125677f.

File: components/AppBar/src/MDCAppBarViewController.m (hunk 3 above)
Comment (fluxminds) [reply components/AppBar/src/MDCAppBarViewController.m]: Addressed in 1b97d553ff0a58d5597bbe7382a338085125677f.

File: components/AppBar/tests/unit/MDCAppBarViewControllerTests.m (hunk 4 above)
Comment (fluxminds) [reply components/AppBar/tests/unit/MDCAppBarViewControllerTests.m]: Addressed in 1b97d553ff0a58d5597bbe7382a338085125677f.

File: components/AppBar/src/MDCAppBarViewController.h (hunk 1 above)
Comment (fluxminds) [reply components/AppBar/src/MDCAppBarViewController.h:41]: Added as additional docs in 1b97d553ff0a58d5597bbe7382a338085125677f.

**Pull request merged**